src/common/store/cache/tracks/sagas.js

```diff
@@ -408,7 +408,17 @@ function* watchFetchCoverArt() {
         cacheActions.update(Kind.TRACKS, [{ id: trackId, metadata: track }])
       )
 
-      const dominantColors = yield call(dominantColor, url)
+      let smallImageUrl = url
+      if (coverArtSize !== SquareSizes.SIZE_150_BY_150) {
```

Comment:
I also don't think this is a super viable long-term solution -- we'll end up fetching two images a decent amt of the time (direct link to track page, etc.) -- it kind of undoes the work around having multiple image sizes.

Do you think we can we do this (or better the 1000x1000 request on opening of the visualizer?)

Or maybe we could check that the size is 480x480 here and only then fetch the dominant colors? 480x480 being the size of the now playing artwork.

I'm ok with merging this change for now since this action is handled async and won't block page loads or anything, but we should try to avoid doing this if we can.

Reply:
I'll try out the fetching on visualizer open in a separate PR

Reply:
agree!